src/app/lib/pages/map_page.dart

```diff
@@ -1,21 +1,904 @@
+// ignore_for_file: prefer_const_constructors
+
 import 'package:flutter/material.dart';
-import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
 
 class MapPage extends StatefulWidget {
   const MapPage({Key? key}) : super(key: key);
 
-  
-
   @override
   MapPageState createState() => MapPageState();
 }
 
 class MapPageState extends State<MapPage> {
+  Set<Marker> _markers = {};
+
+  void _onMapCreated(GoogleMapController controller) {
+    controller.setMapStyle(
+        MapStyle.someLandMarks); //TODO: Allow users to choose their theme
+    setState(() {
+      //here we could maybe loop through all our current events and add to map
+      _markers.add(Marker(
+          markerId: MarkerId("event1"),
+          position: LatLng(43.55103829955488, -79.66262838104547),
+          infoWindow: InfoWindow(
+              title: "Event number 1", snippet: "event number 1 description")));
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-        body: Container(
-            padding: const EdgeInsets.fromLTRB(150.0, 180.0, 0.0, 0.0),
-            child: const Text("Map Page")));
+        body: GoogleMap(
+            onMapCreated: _onMapCreated,
+            markers: _markers,
+            initialCameraPosition: CameraPosition(
+                target: LatLng(43.55103829955488, -79.66262838104547),
+                zoom: 15)));
+  }
+}
+
+class MapStyle {
+  static String noLandMarks = '''
+  [
+  {
+    "featureType": "administrative",
+    "elementType": "geometry",
+    "stylers": [
+      {
+        "visibility": "off"
+      }
+    ]
+  },
+  {
+    "featureType": "poi",
+    "stylers": [
```

Comment:
hmm, these styles written in string seem quite easy to make a mistake. I wonder if there's any way to highlight the syntax